pika/channel.py

```diff
@@ -753,127 +756,134 @@ def open(self):
         self._add_callbacks()
         self._rpc(spec.Channel.Open(), self._on_openok, [spec.Channel.OpenOk])
 
-    def queue_bind(self, callback, queue, exchange,
+    def queue_bind(self,
+                   queue,
+                   exchange,
                    routing_key=None,
-                   nowait=False,
-                   arguments=None):
+                   arguments=None,
+                   callback=None):
         """Bind the queue to the specified exchange
 
-        :param callable callback: The callback to call on Queue.BindOk;
-            MUST be None when nowait=True.
         :param queue: The queue to bind to the exchange
         :type queue: str or unicode
         :param exchange: The source exchange to bind to
         :type exchange: str or unicode
         :param routing_key: The routing key to bind on
         :type routing_key: str or unicode
-        :param bool nowait: Do not wait for a Queue.BindOk
         :param dict arguments: Custom key/value pair arguments for the binding
+        :param callable callback: The callback to call on Queue.BindOk
+        :raises ValueError:
 
         """
-        self._validate_channel_and_callback(callback)
-        replies = [spec.Queue.BindOk] if nowait is False else []
+        self._validate_channel()
+        nowait = self._validate_rpc_completion_callback(callback)
         if routing_key is None:
             routing_key = queue
         return self._rpc(spec.Queue.Bind(0, queue, exchange, routing_key,
                                          nowait, arguments or dict()),
-                         callback, replies)
+                         callback,
+                         [spec.Queue.BindOk] if not nowait else [])
 
-    def queue_declare(self, callback,
-                      queue='',
+    def queue_declare(self,
```

Comment:
docs/examples/tornado_consumer.rst has an example with incorrectly ordered args that won't work: `self._channel.queue_declare(self.on_queue_declareok, queue_name)`